core/src/main/scala/kafka/controller/ControllerEventManager.scala

```diff
@@ -69,7 +69,8 @@ class QueuedEvent(val event: ControllerEvent,
 class ControllerEventManager(controllerId: Int,
                              processor: ControllerEventProcessor,
                              time: Time,
-                             rateAndTimeMetrics: Map[ControllerState, KafkaTimer]) extends KafkaMetricsGroup {
+                             rateAndTimeMetrics: Map[ControllerState, KafkaTimer],
+                             eventQueueTimeTimeoutMs: Long = 60000) extends KafkaMetricsGroup {
```

Comment:
The exact value probably depends on the metric collecting interval. Perhaps we could be a bit conservative with a 5 min timeout?